src/Providers/EUProvider.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Ageras\Sherlock\Providers;
+
+use Ageras\Sherlock\Exceptions\MethodNoImplemented;
+use Ageras\Sherlock\Models\Company;
+use SoapClient;
+use SoapFault;
+use Ageras\Sherlock\Exceptions\SoapClientException;
+
+class EUProvider implements CompanyProviderInterface
+{
+    /**
+     * Service URL.
+     * @var string
+     */
+    protected $serviceUrl = 'http://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl';
+
+    /**
+     * Geo code.
+     * @var string
+     */
+    private $geo_code;
+
+    /**
+     * EUProvider constructor.
+     * @param $geoCode
+     */
+    public function __construct($geoCode)
+    {
+        $this->geo_code = $geoCode;
+    }
+
+    /**
+     * Query builder.
+     * @param $string
+     * @return array
+     */
+    protected function query($string)
+    {
+        $geo_code = strtoupper($this->geo_code);
+        try {
+            $search = new SoapClient($this->serviceUrl);
+            $result = $search->checkVat(['countryCode' => $geo_code, 'vatNumber' => $string]);
+
+            return count($result) > 0 ? $this->formatResult($result) : null;
+        } catch (SoapFault $e) {
+            throw new SoapClientException($e);
+        }
+    }
+
+    /**
+     * Format result.
+     * @param $data
+     * @return array
+     */
+    protected function formatResult($data)
+    {
+        $result[] = new Company([
+            'company_name'                => $data->name,
+            'company_status'              => $data->valid ? Company::COMPANY_STATUS_ACTIVE : Company::COMPANY_STATUS_CEASED,
+            'company_registration_number' => null,
+            'company_vat_number'          => $data->vatNumber,
+            'company_address'             => $this->formatAddress($data->address),
+            'company_city'                => $this->formatCity($data->address),
+            'company_postcode'            => $this->formatPostcode($data->address),
+            'company_phone_number'        => null,
+            'company_email'               => null,
+        ]);
+
+        return $result;
+    }
+
+    /**
+     * Format address.
+     * @param $data
+     * @return null|string
+     */
+    protected function formatAddress($data)
+    {
+        $datas = explode("\n", $data);
+        foreach ($datas as $address) {
+            if (! empty($address)) {
+                return trim($address);
+            }
+        }
+    }
+
+    /**
+     * Format city.
+     * @param $data
+     * @return null|string
+     */
+    protected function formatCity($data)
+    {
+        $data = explode(' ', $data);
```

Comment:
This assumes that the city is the last element, which might not be true in all cases.